api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java

```diff
@@ -62,6 +62,12 @@ public FreeTierBillingService(
     this.clock = clock;
   }
 
+  public boolean isUsingFreeTier(DbWorkspace workspace) {
```

Comment:
thinking out loud: Currently, there's no capability or plan for our support to be able to give someone enough free credits to a customer with a billing account to get through a couple of days while they sort out a billing or config issue with Google? I.e. you can only ever get free credits on a free tier account.

Reply:
That specific implementation solution is not something that's at all supported by GCP billing or by the billing model we're using.

Setting that aside, the way to resolve that use case in the current model would be to switch the user back to free tier. As currently proposed, the user would need to do this manually; or if they wait too long (and data is at risk of deletion), we will automatically switch them back.

Reply:
Our implementation plan is to set user-specific overrides (for both cost and time) which will restore users to the free tier when set.

Reply:
I don't think that matches my understanding of the plan. I think this is all outlined in the synthesis doc, but let me walk through this example to be clear.

Scenario: Workspace's user-provided billing account becomes inactive -> workspace becomes "billing locked"
Resolutions:
1. (if creator still has free credits) owner can switch the workspace to free tier
2. owner can fix the billing account
3. owner can attach a different billing account
4. owner does not resolve billing issues for 2 weeks: workspace reverts to free tier
4a. If the user still has free tier credits, this is now a normal, functional free tier workspace
4b. If the user has no credits, this is still a "billing locked" workspace; but one which is attached to the free tier billing account (and therefore, data will not be deleted)

(4b) is an identical end-state to the easier-to-follow scenario where a user has a free tier workspace and runs out of credits. The workspace is locked, but still incurs some service-level cost to free tier

Reply:
also, overriding their free tier quota (on dollars or time), would not directly effect the free tier status of any of the workspaces, it would just change whether or not they are billing locked.

Reply:
Sorry, I was confused.  I was thinking of the free-tier-only flow.  Your response matches my understanding.